### DISCUSSION TITLE
video stimuli loading very slowly

### DISCUSSION THREAD

**Discussion by rbruno1** [jspsych/jsPsych#2306] [Q&A] [answered]
My stimuli are talking two minutes to load between trials. I am preloading using this code on JsPsych 6.3.1
```
var preload = {
    type: 'preload',
    auto_preload: true 
};
```

**Comment by jodeleeuw** [comment, marked as answer]
Hi @rbruno1,

Are you video files very large? How are you hosting them?

**Reply by rbruno1** [reply]
each are around 1 MB. I am hosting them on cognition. Here is a bit of my code. it is strange, the practice trials are running fine. but then the actual experiment begins and i am waiting eons for each trial.
```
var preload = {
    type: 'preload',
    auto_preload: true 
};

var repeated = false; 

const practice_Clean_1 = {
  timeline: [
  {type: 'video-keyboard-response',
  stimulus: ['PencilSees_Clean_AV_Full_Scale60.mp4'],
  response_allowed_while_playing: false,
  width:1200,
  trial_ends_after_video: true,
  post_trial_gap: 250},
  {    type:'html-button-response',
    stimulus: ["Press <strong>→</strong> to type your answer, or <strong>↺</strong> to replay the sound (you will only be allowed one replay)"],
    choices: ['→', '↺'],
    post_trial_gap: 500,
    on_start: function(trial) {
      if (repeated) {
        trial.choices = ['→'];
      }
    }
  }],
  loop_function: function(data){
    if(data.values()[1].response == 1
    && !repeated){
      repeated = true;
      return true;
    } else {
      repeated = false;
      return false;
    }
  }
};

var POD_AV_Clean_stimuli_Order1 = [
{stimulus: ['PeanutFound_Clean_AV_POD_Scale60.mp4']}, 
{stimulus: ['RulerAsked_Clean_AV_POD_Scale60.mp4']}, 
{stimulus: ['SandalSaw_Clean_AV_POD_Scale60.mp4']}, 
{stimulus: ['BagelSees_Clean_AV_POD_Scale60.mp4']}, 
{stimulus: ['CocoaTalked_Clean_AV_POD_Scale60.mp4']}
];

var POD_AV_Clean_trials_Order1 = {
    timeline: [
    {
        type: 'video-keyboard-response',
        stimulus: jsPsych.timelineVariable('stimulus'),
        choices: jsPsych.NO_KEYS,
        width:1200,
        trial_ends_after_video: true,
        post_trial_gap: 500
    },
    {
        type: 'html-button-response',
        stimulus: [""],
        choices: ['→', '↺'],
        post_trial_gap: 500,
        on_start: function(trial) {
          if (repeated) {
            trial.choices = ['→'];
         }
        }
     }],
     loop_function: function(data){
        if(data.values()[1].response == 1
        && !repeated){
          repeated = true;
          return true;
       }  else {
          repeated = false;
          return false;
    }
  }
};

const POD_AV_Clean_response_Order1 = {
        type: 'survey-text',
        questions: [
          {prompt: ""}
        ],
        post_trial_gap: 500,
        data: {
          task: 'response',
          correct_response: jsPsych.timelineVariable('correct_response')
        },
        on_finish: function(data) {
          data.correct = data.response.Q0.toLowerCase().trim() == data.correct_response;
      },
    }

var POD_AV_Clean_procedure_Order1 = {
  timeline: [POD_AV_Clean_trials_Order1, POD_AV_Clean_response_Order1],
  timeline_variables: POD_AV_Clean_stimuli_Order1,
  repetitions: 1,
  randomize_order: true
}
```

**Reply by becky-gilbert** [reply]
@rbruno1 if you open the Network tab in the browser's developer tools and then load the experiment:
- do you see all of your files loading during the preload trial?
- do you see that a file loads _again_ when a trial starts? (maybe just on the trials with the longer delays)

This information would help to figure out whether the problem is that your preloading isn't working (possibly just for some files), or whether preloading is working but the files aren't loading from the browser's memory cache when the trial starts. Feel free to post your code and/or send a link to your experiment if you'd like us to help check.

**Reply by rbruno1** [reply]
https://lkneqlrb9d.cognition.run/

Select Clean Speech - Order 1 as condition. Thank you for your help!!!

**Reply by rbruno1** [reply]
So i see in the developer tools that i can set 'override_safe_mode' to true. in JsPsych.init. Is there a way to do this in JsPsych 6.3.1?

**Reply by jodeleeuw** [reply]
Yes, you can do this in 6.3.1. I don't think it will help this particular problem because safe mode only applies when jsPsych is loaded on a local disk through a `file://` protocol. Since you are on cognition.run it won't have any effect.

I tried loading your experiment above but it won't load. I think you have syntax errors near jsPsych.init():

![image](https://user-images.githubusercontent.com/595524/140534098-0a110a97-ebf5-42f2-97c6-96728b163067.png)

This is being flagged because the call to `initJsPsych()` has a closing paren , but you are trying to create an object. It should look like this:

```js
const jsPsych = initJsPsych({
    use_webaudio: false,
    override_safe_mode: true,
    on_finish: function(){
        jsPsych.data.displayData();
    }
});
```

But it also looks like you are using version 7.0 (`initJsPsych()`) and 6.3.1 (the plugins are all declared with string types) at the same time. This won't work, so just check to verify that you are being consistent with the version that you've elected to use.

**Reply by rbruno1** [reply]
sorry. I am trying to migrate it to jspsych 7.0. I am in the process of doing that now. That is why it won't load. It should work now

**Reply by rbruno1** [reply]
I ended up just manually preloading. https://urldefense.com/v3/__https://www.jspsych.org/6.3/overview/media-preloading/*manual-preloading__;Iw!!OlavHw!u_Z4DaOct4ISNm_PGPryz0oLZTFVESgVrNUnDV9OKNXHbbEAeGXS4OycpRWY6Hu-Kw$
